File: mini-coupang/seller-service/src/main/java/com/deepblog/seller/product/repository/ProductRepository.java
package com.deepblog.seller.product.repository;

import com.deepblog.seller.product.entity.Product;
import org.springframework.data.jpa.repository.JpaRepository;

public interface ProductRepository extends JpaRepository<Product, Long> {

    boolean existsBySellerIdAndSku(Long sellerId, String sku);
}

File: services/mini-coupang/src/main/kotlin/com/deepblog/minicoupang/domain/product/storage/ProductRepository.kt
package com.deepblog.minicoupang.domain.product.storage

import com.deepblog.minicoupang.domain.product.Product
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface ProductRepository : JpaRepository<Product, Long> {
    fun findBySellerId(sellerId: Long): List<Product>
}
